crates/indexer/src/models/coin_models/account_transactions.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+// This is required because a diesel macro makes clippy sad
+#![allow(clippy::extra_unused_lifetimes)]
+#![allow(clippy::unused_unit)]
+
+use crate::{
+    models::{
+        token_models::v2_token_utils::ObjectWithMetadata, user_transactions::UserTransaction,
+    },
+    schema::account_transactions,
+    util::standardize_address,
+};
+use aptos_api_types::{DeleteResource, Event, Transaction, WriteResource, WriteSetChange};
+use field_count::FieldCount;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+
+pub type AccountTransactionPK = (String, i64);
+
+#[derive(Debug, Deserialize, FieldCount, Identifiable, Insertable, Serialize)]
+#[diesel(primary_key(account_address, transaction_version))]
+#[diesel(table_name = account_transactions)]
+pub struct AccountTransaction {
+    pub transaction_version: i64,
+    pub account_address: String,
+}
+
+impl AccountTransaction {
+    /// This table will record every transaction that touch an account which could be
+    /// a user account, an object, or a resource account.
+    /// We will consider all transactions that modify a resource or event associated with a particular account.
+    /// We will do 1 level of redirection for now (e.g. if it's an object, we will record the owner as account address).
+    /// We will also consider transactions that the account signed or is part of a multi sig / multi agent.
+    /// TODO: recursively find the parent account of an object
+    /// TODO: include table items in the detection path
+    pub fn from_transaction(
+        transaction: &Transaction,
+    ) -> anyhow::Result<HashMap<AccountTransactionPK, Self>> {
+        let (events, wscs, signatures, txn_version) = match transaction {
```

Comment:
This looks like a fair amount of common code. Is there a common library to call to do this?

Reply:
This will need to be cleaned up but for now I don't have time to refactor all that code.